application/src/main/java/com/softserve/lv460/application/service/HomeService.java

```diff
@@ -0,0 +1,84 @@
+package com.softserve.lv460.application.service;
+
+import com.softserve.lv460.application.dto.home.HomeRequest;
+import com.softserve.lv460.application.dto.home.HomeResponse;
+import com.softserve.lv460.application.dto.location.LocationResponse;
+import com.softserve.lv460.application.entity.Home;
+import com.softserve.lv460.application.repository.HomeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class HomeService {
+  @Autowired
+  private HomeRepository homeRepository;
+  @Autowired
+  private LocationService locationService;
+
+  public Home create(HomeRequest hR) {
+    Optional<Home> h=homeRepository.findByAddressaLike(hR.getAddressa());
```

Comment:
"h" is not a good variable name.

Reply:
also, you can take a look at one of mappers implementation for dto <-> entity conversions